### ISSUE TITLE
Change downloads, favorites and acccounts storage to json instead of databases.

### ISSUE THREAD

**Issue by riderkick** [riderkick/FMD#1019]
Currently we store the data with SQLite databases.
The disadvantage is, it's not flexible enough when we need to modify the fields (add new field). We need to recreate an entire table to do that. And check between different table may become a problem in the future.
Furthermore we load everything when loading the databases. So, we don't really get the advantages of using the databases. And we never use sql query anyway.
To be able to make a transition then we gotta keep the current databases related codes.
But at some point I would like to remove all of that. Keeping unused codes is not a good practice.
Or I can just remove it right away and provide users an external application to make a conversion.

Let's discuss the process here.

**Comment by Tmp341** [comment]
You wanna remove databases?

**Comment by riderkick** [comment]
Yes, the databases of downloads, favorites and accounts. We never really use it as databases. All data fully loaded at startup.

**Comment by ralvarador** [comment]
but, is there any structure that could save resources + be easy to mantain?
I mean, at least with DB you're sure that it's quite difficult to mess-up the data that are managed by the APP (or at least no-one wants put it its hands in the DB). (I'm already seeing the developers saying: do you know what we are talking about?)

**Comment by riderkick** [comment]
There is a cost between transactions when using databases. Because we load everything anyway, read from the file directly would be faster.

**Comment by kmvi** [comment]
i started moving favorites to json: https://github.com/kmvi/FMD/tree/favorites-json
type of `DownloadedChapterList` should be changed to array/stringlist, there is no point in storing it as a string

**Comment by ChocolateOtaku** [comment]
Will there be a migration of favorites from SqliteDB to JSON or will favorites be lost?

**Comment by kmvi** [comment]
Favorites will be automatically converted to json (both sqlite db and old ini format supported)

**Comment by ChocolateOtaku** [comment]
When will favorites in JSOn be ready?
Is there even active development ongoing?
At the beginning of the year there was a new release almost any 3 days and now...
https://github.com/riderkick/FMD/graphs/commit-activity

**Comment by Tmp341** [comment]
@ChocolateOtaku It was when @riderkick wasn't MIA.

**Comment by ralvarador** [comment]
Also, when LUA was implemented... Any change on any server or reader imply just modify the LUA module(s) and not the whole Software, and the modules are automatically updatded. This - mostly, solve all request to the project (except a functionality issue, fix or add a feature)
:shrug:
This is a really useful tool.
